org.eclipse.scout.rt.dataobject.test/src/main/java/org/eclipse/scout/rt/dataobject/testing/AbstractTypeVersionClassNamingTest.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2010, 2025 BSI Business Systems Integration AG
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.scout.rt.dataobject.testing;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+
+import org.eclipse.scout.rt.dataobject.AbstractTypeVersion;
+import org.eclipse.scout.rt.dataobject.ITypeVersion;
+import org.eclipse.scout.rt.platform.BEANS;
+import org.eclipse.scout.rt.platform.util.StringUtility;
+import org.eclipse.scout.rt.testing.platform.runner.PlatformTestRunner;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(PlatformTestRunner.class)
+public abstract class AbstractTypeVersionClassNamingTest {
+
+  @Test
+  public final void testTypeVersions() {
+    List<String> malformedTypeVersionClassNames = collectMalformedTypeVersionClassNames();
+    if (!malformedTypeVersionClassNames.isEmpty()) {
+      Assert.fail("Malformed type version class names found.\nType version class names must follow the following pattern: "
+          + "<Namespace>_<version1>[_<version2>[_<versionN>]][__<comment>] where <Namespace> must be called after the respective scout namespace with all "
+          + "letters in lower case except for the first letter, which must be capitalized.\n- " + String.join("\n- ", malformedTypeVersionClassNames));
+    }
+  }
+
+  protected abstract String getPackageNamePrefix();
+
+  protected Set<Class<ITypeVersion>> getExclusionList() {
+    return Collections.emptySet();
```

Comment:
I would return a mutable set here, so that you can just use the supercall to add additional exclusions.

```suggestion
    return CollectionUtility.emptyHashSet();
```